Regression in k0s ostests (0.17.5 -> 0.17.7)

https://github.com/k0sproject/k0s/actions/runs/9015323184

This is an example from Debian 12:

```text
│ level=info msg="==> Running phase: Upload k0s binaries to hosts"
│ level=info msg="[ssh] 54.234.200.191:22: uploading k0s binary from
│ /home/runner/work/k0s/k0s/.cache/k0s to
│ /usr/local/bin/k0s.tmp.1715247393191793148"
│ level=info msg="[ssh] 54.224.68.241:22: uploading k0s binary from
│ /home/runner/work/k0s/k0s/.cache/k0s to
│ /usr/local/bin/k0s.tmp.1715247393191803151"
│ level=info msg="[ssh] 34.235.141.247:22: uploading k0s binary from
│ /home/runner/work/k0s/k0s/.cache/k0s to
│ /usr/local/bin/k0s.tmp.1715247393191819104"
│ level=info msg="[ssh] 52.55.26.96:22: uploading k0s binary from
│ /home/runner/work/k0s/k0s/.cache/k0s to
│ /usr/local/bin/k0s.tmp.1715247393191831547"
│ level=info msg="[ssh] 35.173.196.244:22: uploading k0s binary from
│ /home/runner/work/k0s/k0s/.cache/k0s to
│ /usr/local/bin/k0s.tmp.1715247393191850182"
│ level=info msg="* Running clean-up for phase: Acquire exclusive host lock"
│ level=info msg="==> Apply failed"
│ level=fatal msg="apply failed - log file saved to
│ /home/runner/.cache/k0sctl/k0sctl.log: failed on 5 hosts:\n - [ssh]
│ 54.234.200.191:22: upload k0s binary: invalid path: open remote file
│ /usr/local/bin/k0s.tmp.1715247393191793148 for writing: open
│ /usr/local/bin/k0s.tmp.1715247393191793148: command failed: client exec:
│ ssh session wait: Process exited with status 1\n - [ssh] 54.224.68.241:22:
│ upload k0s binary: invalid path: open remote file
│ /usr/local/bin/k0s.tmp.1715247393191803151 for writing: open
│ /usr/local/bin/k0s.tmp.1715247393191803151: command failed: client exec:
│ ssh session wait: Process exited with status 1\n - [ssh] 52.55.26.96:22:
│ upload k0s binary: invalid path: open remote file
│ /usr/local/bin/k0s.tmp.1715247393191831547 for writing: open
│ /usr/local/bin/k0s.tmp.1715247393191831547: command failed: client exec:
│ ssh session wait: Process exited with status 1\n - [ssh] 34.235.141.247:22:
│ upload k0s binary: invalid path: open remote file
│ /usr/local/bin/k0s.tmp.1715247393191819104 for writing: open
│ /usr/local/bin/k0s.tmp.1715247393191819104: command failed: client exec:
│ ssh session wait: Process exited with status 1\n - [ssh] 35.173.196.244:22:
│ upload k0s binary: invalid path: open remote file
│ /usr/local/bin/k0s.tmp.1715247393191850182 for writing: open
│ /usr/local/bin/k0s.tmp.1715247393191850182: command failed: client exec:
│ ssh session wait: Process exited with status 1"
│ 
╵
Error: Process completed with exit code 1.
```

--- a/phase/upload_k0s.go
+++ b/phase/upload_k0s.go
@@ -67,11 +67,11 @@ func (p *UploadK0s) uploadBinary(h *cluster.Host) error {
 	}
 
 	log.Infof("%s: uploading k0s binary from %s to %s", h, h.UploadBinaryPath, tmp)
-	if err := h.Upload(h.UploadBinaryPath, tmp); err != nil {
+	if err := h.Upload(h.UploadBinaryPath, tmp, exec.Sudo(h)); err != nil {
 		return fmt.Errorf("upload k0s binary: %w", err)
 	}
 
-	if err := h.Configurer.Touch(h, tmp, stat.ModTime()); err != nil {
+	if err := h.Configurer.Touch(h, tmp, stat.ModTime(), exec.Sudo(h)); err != nil {
 		return fmt.Errorf("failed to touch %s: %w", tmp, err)
 	}
 	if err := h.Execf(`chmod +x "%s"`, tmp, exec.Sudo(h)); err != nil {

